core/bootstrap/src/main/java/org/wildfly/swarm/bootstrap/util/TempFileManager.java

```diff
@@ -109,7 +118,9 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e)
         return true;
     }
 
-    private Set<File> registered = Collections.newSetFromMap(new ConcurrentHashMap<>());
+    private final Set<File> registered = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    private final AtomicReference<File> explodedApplicationArtifact = new AtomicReference<>();
```

Comment:
Not quite sure why this is a `AtomicReference` -- looking at what you do with it, making it `volatile` should be enough.

Reply:
Right, `volatile` should be enough. I'm just used to use `AtomicReference` wrapper in similar scenarios as the "contract/API" is imho more clear and "it does not hurt" ;-). But if you feel like it's inappropriate here we can use `volatile` instead...

Reply:
It's not inappropriate, it's just my habits are the other way around :-)